### PR TITLE
feat: Output now displays information in a more concise and informative manner

### PR DESCRIPTION
**Pull request by jsamar17**
In addition to this, 2 helper functions getNumberOfTransitiveDependencies and squashDetections were written

### CODE REVIEW

File: src/output-util.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import {PackageTree, PointOfInterest} from './package-tree';
+
+export function getNumberOfTransitiveDetections(
+    packageTree: PackageTree<PointOfInterest[]>,
+    countedDependencies: Map<string, PackageTree<PointOfInterest[]>>): number {
+  let totalDependencies = packageTree.data.length;
```
Comment (kjin) [inline src/output-util.ts]: This seems like should be `totalDetections`, right?

File: src/output-util.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Comparison function to sort Points of Interest by type
+function compare(a: PointOfInterest, b: PointOfInterest): number {
+  if (a.type < b.type) {
+    return -1;
+  }
+  if (a.type > b.type) {
+    return 1;
+  }
+  return 0;
+}
```
Comment (kjin) [inline src/output-util.ts]: Newline

File: src/output-util.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      totalDependencies +=
+          getNumberOfTransitiveDetections(dep, countedDependencies);
+    }
+  });
+  return totalDependencies;
+}
+
+export function squashDetections(packageTree: PackageTree<PointOfInterest[]>):
+    Map<string, number> {
+  const sortedDataArray = packageTree.data.sort(compare);
```
Comment (kjin) [inline src/output-util.ts]: This mutates `packageTree.data`. You can copy the array before sorting it like `packageTree.data.slice(0).sort(compare)` or similar

File: src/output.ts [hunk 4]
```diff
@@ -47,15 +48,35 @@ function flattenPackageTree(
 function output(packageTrees: Array<PackageTree<PointOfInterest[]>>): string {
   const arrOfStrings: string[] = [];
   packageTrees.forEach((packageTree) => {
-    arrOfStrings.push(
-        `Name: ${packageTree.name}, Version: ${packageTree.version}`);
+    arrOfStrings.push(`${packageTree.name} ${packageTree.version} Detections: ${
+        packageTree.data.length} Immediate ${
+        getNumberOfTransitiveDetections(
+            packageTree,
+            new Map<string, PackageTree<PointOfInterest[]>>())} Transitive`);
```
Comment (kjin) [inline src/output.ts]: I would suggest defining an inner `getNumberOfTransitiveDetectionsRec` function and initializing the `Map` in `getNumberOfTransitiveDetections`. This helps clean up the function signature, as users don't have to pass in their own map which ends up becoming an implementation detail.

File: src/output.ts [hunk 5]
```diff
@@ -47,15 +48,35 @@ function flattenPackageTree(
 function output(packageTrees: Array<PackageTree<PointOfInterest[]>>): string {
   const arrOfStrings: string[] = [];
   packageTrees.forEach((packageTree) => {
-    arrOfStrings.push(
-        `Name: ${packageTree.name}, Version: ${packageTree.version}`);
+    arrOfStrings.push(`${packageTree.name} ${packageTree.version} Detections: ${
+        packageTree.data.length} Immediate ${
```
Comment (kjin) [inline src/output.ts]: Comma after `Immediate` would be clearer?

**Review by jplaisted — APPROVED** [review]

**Pull request merged**